### PR TITLE
Sales webhooks

### PR DESCRIPTION
**Pull request by kuchichan**
I want to merge this change because it introduces webhooks for C(R)UD for Sales. Webhooks calculates differences in catalogue fields.  

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/discount/tests/test_discount.py [hunk 1]
```diff
@@ -1,14 +1,17 @@
 from datetime import timedelta
-from unittest.mock import ANY
+from unittest.mock import ANY, patch
 
 import graphene
 import pytest
 from django.utils import timezone
 from django_countries import countries
 
+from saleor.graphql.discount.mutations import convert_catalogue_info_to_global_ids
```
Comment (korycins) [inline saleor/graphql/discount/tests/test_discount.py:9]: Convert it to relative

File: saleor/plugins/base_plugin.py [hunk 2]
```diff
@@ -379,6 +380,29 @@ def order_confirmed(self, order: "Order", previous_value: Any):
         """
         return NotImplemented
 
+    def sale_created(self, sale: "Sale", current_catalogue, previous_value: Any):
```
Comment (korycins) [inline saleor/plugins/base_plugin.py:383]: Are we able to add typing to these methods? For someone who wants to use it in own plugin, this woud be very helpfull

**Review by korycins — COMMENTED** [review]
@kuchichan Looks good, but it would be great if you could add some tests also in `test_manager.py`.

File: saleor/plugins/base_plugin.py (hunk 2 above)
Comment (kuchichan) [reply saleor/plugins/base_plugin.py:383]: Yes, ofc, I was thinking even about named type like `CatalogueInfo = Dict[str,  Set[int]]` and `NodeCatalogueInfo = Dict[str, Set[str]]`. What do you think?

File: saleor/graphql/discount/tests/test_discount.py (hunk 1 above)
Comment (kuchichan) [reply saleor/graphql/discount/tests/test_discount.py:9]: Done.

File: saleor/plugins/base_plugin.py (hunk 2 above)
Comment (kuchichan) [reply saleor/plugins/base_plugin.py]: Typing added.

File: saleor/plugins/webhook/plugin.py [hunk 3]
```diff
@@ -89,24 +90,31 @@ def order_updated(self, order: "Order", previous_value: Any) -> Any:
...
-    def sale_created(self, sale: "Sale", current_catalogue, previous_value: Any) -> Any:
+    def sale_created(
+        self, sale: "Sale", current_catalogue: "NodeCatalogueInfo", previous_value: Any
+    ) -> Any:
         if not self.active:
             return previous_value
         sale_data = generate_sale_payload(
             sale, previous_catalogue=None, current_catalogue=current_catalogue
         )
+        print(sale_data)
```
Comment (korycins) [inline saleor/plugins/webhook/plugin.py]: 🚔 🚔 🚔

Comment (kuchichan) [reply saleor/plugins/webhook/plugin.py]: 🐴

**Review by korycins — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/discount/tests/test_discount.py (hunk 1 above)
Comment (fowczarek) [reply saleor/graphql/discount/tests/test_discount.py:9]: :policeman:

Comment (kuchichan) [reply saleor/graphql/discount/tests/test_discount.py]: How :( Done for real.

File: saleor/webhook/payloads.py [hunk 4]
```diff
@@ -223,6 +226,61 @@ def generate_order_payload(order: "Order"):
     return order_data
 
 
+def calculate_added(
```
Comment (maarcingebala) [inline saleor/webhook/payloads.py:229]: When reading through this module, `calculate_added` and `calculate_removed` seem a bit unexpected names among all these payload generators, maybe we should make them "private"?

**Comment by maarcingebala** [comment]
@kuchichan We should update the changelog as well (3.1 section).

File: saleor/webhook/payloads.py (hunk 4 above)
Comment (kuchichan) [reply saleor/webhook/payloads.py]: Done.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**